### ISSUE TITLE
Questions about cal_bl_loss?

### ISSUE THREAD

**Issue by WenxiongLiao** [laihuiyuan/Pre-trained-formality-transfer#7]
Hi, cal_bl_loss  calls NLTK method sentence_bleu, but sentence_bleu cannot calculate the gradient. I want to ask how cal_bl_loss  performs gradient descent?

**Comment by laihuiyuan** [comment]
Hi! The gradient is obtained via the log-derivative trick for policy gradients (see cal_reward_loss). Thank you!
